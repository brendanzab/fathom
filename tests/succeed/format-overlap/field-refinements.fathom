let number = overlap {
    u <- u32be where u32_gte u 2,
    s <- s32be,
};

let _ : number.Repr -> { u : U32, s : S32 } =
    fun n => n;

{}
