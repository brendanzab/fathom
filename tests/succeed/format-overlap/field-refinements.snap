stdout = '''
let number : Format = overlap { u <- u32be where u >= (2 : U32), s <- s32be };
let _ : number.Repr -> { u : U32, s : S32 } = fun n => n;
() : ()
'''
stderr = ''
